app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -48,17 +48,13 @@ def parent_cluster
   alias ems_cluster parent_cluster
 
   def disconnect_storage(_s = nil)
-    unless active?
-      return
-    end
-    vm_disks = collect_disks
+    return unless active?
 
-    if vm_disks.blank?
-      storage = nil
-    else
-      vm_storages = ([storage] + storages).compact.uniq
-      storage = vm_storages.select { |store| !vm_disks.include?(store.ems_ref) }
-    end
+    vm_storages = ([storage] + storages).compact.uniq
+    return if vm_storages.empty?
+
+    vm_disks = collect_disks
```

Comment:
@masayag No, if the disk is removed it will be not listed. According to my understanding of the code disk.storage always will have a value.